server/service/datapool/validate_data_handler.go

```diff
@@ -56,7 +58,21 @@ func (svc *dataPoolService) handleValidateData(w http.ResponseWriter, r *http.Re
 	// make dataHash
 	dataHash := crypto.Hash(jsonInput)
 
-	// TODO encrypt and store data
+	dataWithAES256, err := crypto.EncryptDataWithAES256(svc.DataEncKey, nil, jsonInput)
+	if err != nil {
+		log.Error("failed to make encrypted data: ", err)
+		http.Error(w, "failed to make encrypted data", http.StatusInternalServerError)
+		return
+	}
+
```

Comment:
Now the `additional` parameter is nil. It will be changed in future.

Reply:
What will the `additional` be used for?

Reply:
The `additional` is something like a secondary key especially for authentications, aka. `header` in https://en.wikipedia.org/wiki/Authenticated_encryption. That can be used for various purposes. Unlike the `nonce` that must be shared with a encrypter and a decrypter, the `additional` must not be shared.
For us, I think the `additional` is not mandatory for now.